lib/round.rb

```diff
@@ -0,0 +1,145 @@
+class Round
+
+  attr_accessor :deck, :turns
+
+  def initialize(deck)
+    @deck = deck
+    @turns = []
+  end
+
+  def current_card
+    deck.cards.first
+  end
+
+  def take_turn(guess)
+    turn = Turn.new(guess, deck.cards.first)
+    turns << turn
+    deck.cards.rotate!
+    return turn
+  end
+
+  def correct?
+    round.turns.first.guess == round.turns.fist.card.answer
+  end
+
+  def number_correct
+    num_correct = 0
+    turn_index = 0
+    until turn_index == @turns.count
+      turns.each do |turn|
+        if turns[turn_index].guess == turns[turn_index].card.answer
+          num_correct += 1
+          turn_index += 1
+        else
+          turn_index += 1
+        end
+      end
+    end
+    num_correct
+  end
+
+  def number_correct_by_category(category)
+    num_correct = 0
+    turn_index = 0
+    until turn_index == @turns.count
+      @turns.each do |turn|
```

Comment:
I think this is slightly more code than we need, though it does work! 
You needed to go through the `turn_index` to access the correct turn, so that's why you have `until turn_index == @turns.count` and then you're incrementing the `turn_index` below this. However, you're also doing `@turns.each` - there is a method called `.each_with_index` that provides you the same functionality as `.each`, but a built-in index as well. 
So your method could look like this: 
```
@turns.each_with_index do |turn, index| 
   if turn.card.category == category && #turn guess is correct
   ...
```

Reply:
In fact, I'm not quite sure that we need an index here...? Maybe we do and I just don't see it. In any case, the code does work, just might be a little too engineered.